src/Streamly/Internal/Data/Stream/IsStream/Generate.hs

```diff
@@ -506,10 +506,15 @@ fromFoldableM = Prelude.foldr consM K.nil
 -- @since 0.4.0
 {-# INLINE_EARLY fromListM #-}
 fromListM :: (MonadAsync m, IsStream t) => [m a] -> t m a
-fromListM = fromStreamD . D.fromListM
+fromListM = fromFoldableM
 {-# RULES "fromListM fallback to StreamK" [1]
     forall a. D.toStreamK (D.fromListM a) = fromFoldableM a #-}
 
+{-# RULES "unfoldrM serial" fromListM = fromListMSerial #-}
```

Comment:
1. typo - change the rule name.
2. Did you check the serial fromListM performance? It should remain unchanged if the rule is working.

Reply:
1 - Done
2 - Yes, the performance is same.